### PR TITLE
1078 settings implement unit tests

### LINKED ISSUE #1078: [Settings] Implement Unit Tests
1. **Settings Component:**

- [x] When the component initializes, it should set up the default active tab based on the last tab stored in local storage.

2. **Admin API Component:**

- [x] should initialize the form with an empty "apicall" control
- [x] should update the form control value when the input value changes
- [x] should call send method when the button is clicked
- [x] should display the ack message when ack is set
- [x] should display the error message when error is set

3. **Countries / Create Country Component:**

- [x] should initialize the form with empty controls
- [x] should update the form control values when input values change
- [x] should call CountryService.create and close the modal when the save button is clicked
- [x] should display error message when there is an error
- [x] should dismiss the modal when dismiss button is clicked

4. **Countries / Edit Country Component:**

- [x] should initialize the form with country data on ngOnInit
- [x] should call CountryService.update and close the modal when the save button is clicked
- [x] should display error message when there 
- [x] should dismiss the modal when dismiss button is clicked

5. **Countries / Search Country Component:**

- [x] When the component initializes, the search form should be initialized with default values.

6. **Education Levels / Create Education Level Component:**

- [x] should initialize form with required fields
- [x] should call onSave and close modal when education level is successfully created
- [x] should handle error when education level creation fails
- [x] should dismiss modal when dismiss is called

7. **Education Levels / Edit Education Level Component:**

- [x] should initialize form with education level data
- [x] should call onSave and close modal when education level is successfully updated
- [x] should handle error when education level update fails
- [x] should dismiss modal when dismiss is called

8. **Education Levels / Search Education Level Component:**

- [x] When the component initializes, the education major form should be initialized and populated with data from the server.

9. **Education Majors / Create Education Major Component:**

- [x] should initialize form with required fields
- [x] should call onSave and close modal when education major is successfully created
- [x] should handle error when education major creation fails
- [x] should dismiss modal when dismiss is called

10. **Education Majors / Edit Education Major Component:**

- [x] should initialize form with education major data
- [x] should call onSave and close modal when education major is successfully updated
- [x] should handle error when education major update fails
- [x] should dismiss modal when dismiss is called

11. **Education Majors / Search Education Major Component:**

- [x] Verify that the importTranslations() method imports translations for education majors.

12. **External Links / Create External Link Component:**

- [x] should load saved lists on initialization
- [x] should call onSave and close modal when external link is successfully created
- [x] should handle error when creating external link fails
- [x] should dismiss modal when dismiss is called

13. **External Links / Edit External Link Component:**

- [x] should initialize form with saved list details
- [x] should load saved lists on initialization
- [x] should call onSave and close modal when external link is successfully updated
- [x] should handle error when updating external link fails
- [x] should dismiss modal when dismiss is called

14. **External Links / Search External Link Component:**

- [x] Verify that the form is initialized correctly with the required form controls.

15. **Help Links / Create Help Link Component:**

- [x] Ensure that users can update an existing help link using the modal form.

16. **Industries / Create Industry Component:**

- [x] should initialize form with empty values
- [x] should call onSave and close modal when industry is successfully created
- [x] should handle error when creating industry fails
- [x] should dismiss modal when dismiss is called

17. **Language Levels / Create Language Levels Component:**

- [x] should initialize form with Language Levels data
- [x] should call onSave and close modal when language level is successfully created
- [x] should handle error when creating language level fails
- [x] should dismiss modal when dismiss is called

18. **Languages / Create Languages Component:**

- [x] should call onSave and close modal when system language is successfully added
- [x] should handle error when adding system language fails
- [x] should dismiss modal when dismiss is called

19. **Occupations / Create Occupation Component:**

- [x] Ensure that the form is initialized correctly.

20. **Occupations / Create Partner Component:**

- [x] Ensure that the form is initialized correctly.

21. **Tasks / Search Task Component:**

- [x] Verify that the component successfully retrieves a list of tasks based on search criteria.

22. **Translations / Drop Down Translation  Component:**

- [x] Verify that the system languages are fetched successfully.

23. **General / General Translation Component:**

- [x] Test filtering translations by language.

24. **Users / Create Users Component:**

- [x] Validation errors should be triggered for required fields (such as email, username, first name, last name, and role), preventing form submission.

25. **Partners**
- [x] **Create Update Partner Component**
- [x] **Search Partner Component**

### PR DESCRIPTION
**Pull request by EhsanEhrari**

### CODE REVIEW

File: ui/admin-portal/src/app/components/settings/users/create-update-user/create-update-user.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+      partnerId: 1,
+      status: 'active',
+      role: 'user',
+      jobCreator: false,
+      approverId: null,
+      purpose: '',
+      sourceCountries: [],
+      readOnly: false,
+      usingMfa: true,
+      password: 'password'
```
Comment (sadatmalik) [inline ui/admin-portal/src/app/components/settings/users/create-update-user/create-update-user.component.spec.ts]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected credentials hardcoded in your code, which might allow unauthorized users to access your account. We recommend you to store your credentials outside of the code in a configuration file, a database, or a management service for secrets and retrieve from it. For more information, see [CWE-259](https://cwe.mitre.org/data/definitions/259.html) and [CWE-798](https://cwe.mitre.org/data/definitions/798.html).



Suggested remediation:
Do not hardcode password or access key. Instead, try to fetch it from environment to deny unauthorized users to access your account.
```
@@ -147,3 +147,3 @@
       usingMfa: true,
-      password: &#39;password&#39;
+      password: process.env.ENV_VARIABLE
     });

```

File: ui/admin-portal/src/app/components/settings/users/create-update-user/create-update-user.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
```
Comment (sadatmalik) [inline ui/admin-portal/src/app/components/settings/users/create-update-user/create-update-user.component.spec.ts]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected credentials hardcoded in your code, which might allow unauthorized users to access your account. We recommend you to store your credentials outside of the code in a configuration file, a database, or a management service for secrets and retrieve from it. For more information, see [CWE-259](https://cwe.mitre.org/data/definitions/259.html) and [CWE-798](https://cwe.mitre.org/data/definitions/798.html).



Suggested remediation:
Do not hardcode password or access key. Instead, try to fetch it from environment to deny unauthorized users to access your account.
```
@@ -185,3 +185,3 @@
       usingMfa: true,
-      password: &#39;password&#39;
+      password: process.env.ENV_VARIABLE
     });

```

File: ui/admin-portal/src/app/components/settings/users/create-update-user/create-update-user.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
```
Comment (sadatmalik) [inline ui/admin-portal/src/app/components/settings/users/create-update-user/create-update-user.component.spec.ts]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected credentials hardcoded in your code, which might allow unauthorized users to access your account. We recommend you to store your credentials outside of the code in a configuration file, a database, or a management service for secrets and retrieve from it. For more information, see [CWE-259](https://cwe.mitre.org/data/definitions/259.html) and [CWE-798](https://cwe.mitre.org/data/definitions/798.html).



Suggested remediation:
Do not hardcode password or access key. Instead, try to fetch it from environment to deny unauthorized users to access your account.
```
@@ -116,3 +116,3 @@
       usingMfa: true,
-      password: &#39;password&#39;
+      password: process.env.ENV_VARIABLE
     });

```

**Review by samschlicht — APPROVED** [review]
Approved

**Pull request merged**